### DISCUSSION TITLE
Automation through API

### DISCUSSION THREAD

**Discussion by colink-snow** [TomBursch/kitchenowl#200] [General]
Don't see any documentation or questions regarding it, is it possible to automate lists via a REST API? I'd like to integrate it into a voice assistant.

**Comment by TomBursch** [comment]
You can definitely do that, however the API documentation is not great, this is high on my list but not something I've come around to yet. In your user profile (profile tab -> settings -> click on your profile at the top -> sessions -> long-lived tokens) you can create long-lived tokens, that should be the easiest way to use the API. This will give you a Bearer token that you can use in your automation to authenticate yourself. The endpoint you most likely want to use is [here](https://github.com/TomBursch/kitchenowl-backend/blob/383d0964dc039b6bd3fadd5f7d3411b3aad1b6ff/app/controller/shoppinglist/shoppinglist_controller.py#L175C41-L175C41) (`api/shoppinglist/<int:id>/add-item-by-name`) and the JSON body format your post request should follow can be seen [here](https://github.com/TomBursch/kitchenowl-backend/blob/main/app/controller/shoppinglist/schemas.py#L4C29-L4C29). If you can read python code everything should not be too hard to understand, if you have any questions feel free to ask them here or on the matrix chat. The shopping list API is pretty stable, so don't expect any changes any time soon. I can let you know once the swagger documentation is up.
